### PR TITLE
Query enable by rule

### PR DESCRIPTION
**Pull request by wburns**
Only allows for query operations to be performed on rules that have query enabled. If not enabled the query operation is just plainly ignored. Also if a user configures a rule with querying without a backend query provider it will error.

### CODE REVIEW

**Comment by wburns** [comment]
Relies upon https://github.com/gingersnap-project/cache-manager/pull/18

File: manager/src/main/java/io/gingersnapproject/Caches.java [hunk 1]
```diff
@@ -26,6 +29,9 @@ public class Caches {
    @Inject
    DatabaseHandler databaseHandler;
 
+   @Inject
+   QueryHandler queryHandler;
```
Comment (fax4ever) [inline manager/src/main/java/io/gingersnapproject/Caches.java]: I think that we cannot have a circular dependency here (Caches <--> QueryHandler), maybe it is better to keep to inject the SearchBackend here

File: manager/src/main/java/io/gingersnapproject/search/SearchBackend.java [hunk 2]
```diff
@@ -1,14 +1,12 @@
 package io.gingersnapproject.search;
 
-import org.infinispan.commons.dataconversion.internal.Json;
-
 import io.smallrye.mutiny.Uni;
 
 public interface SearchBackend {
 
-   Uni<String> mapping(String indexName);
```
Comment (fax4ever) [inline manager/src/main/java/io/gingersnapproject/search/SearchBackend.java:9]: Good idea to hide the mapping, as we said this is not functionality

**Review by fax4ever — COMMENTED** [review]
Nice job! Just a comment about the ci failure

**Comment by fax4ever** [comment]
Here is an idea: https://github.com/gingersnap-project/cache-manager/commit/e385d58840e291fcbb45841e4b88f0e7d2f95cad

File: manager/src/main/java/io/gingersnapproject/Caches.java (hunk 1 above)
Comment (wburns) [reply manager/src/main/java/io/gingersnapproject/Caches.java]: Yeah, I didn't find time yesterday to fix this, yet.

**Comment by wburns** [comment]
> Here is an idea: [e385d58](https://github.com/gingersnap-project/cache-manager/commit/e385d58840e291fcbb45841e4b88f0e7d2f95cad)

Nice, yes this is precisely what I was thinking, but didn't get time to get it. Thanks!

**Comment by wburns** [comment]
> > Here is an idea: [e385d58](https://github.com/gingersnap-project/cache-manager/commit/e385d58840e291fcbb45841e4b88f0e7d2f95cad)
> 
> Nice, yes this is precisely what I was thinking, but didn't get time to get it. Thanks!

Added your commit to my PR.

**Comment by wburns** [comment]
I need to revamp the tests a bit now, since it is not using the proper rule any longer and causes test failures.

**Comment by fax4ever** [comment]
Merged, thanks @wburns!

**Pull request merged**